app/src/main/java/at/roteskreuz/stopcorona/model/repositories/ExposureNotificationRepository.kt

```diff
@@ -198,14 +198,20 @@ class ExposureNotificationRepositoryImpl(
     }
 
     override fun refreshExposureNotificationAppRegisteredState() {
-        exposureNotificationClient.isEnabled
-            .addOnSuccessListener { enabled: Boolean ->
-                frameworkEnabledState.onNext(enabled)
-            }
-            .addOnFailureListener {
-                Timber.e(SilentError(it))
-                frameworkEnabledState.onNext(false)
-            }
+        try {
+            exposureNotificationClient.isEnabled
+                .addOnSuccessListener { enabled: Boolean ->
+                    frameworkEnabledState.onNext(enabled)
+                }
+                .addOnFailureListener {
+                    Timber.e(SilentError(it))
+                    frameworkEnabledState.onNext(false)
+                }.addOnCompleteListener {
+                    Timber.d("it completed")
+                }
+        } catch (ex: java.lang.Exception) {
```

Comment:
please use kotlin exception instead

Reply:
@JustusKlawischS2 this is a draft PR with a Work in Progress commit. please don´t review or check it out 😇